app/controllers/deploys_controller.rb

```diff
@@ -60,9 +47,11 @@ def recent
   #   * status (what is the status of this job failed|running| etc)
 
   def search
-    if (params[:status] && !Job.valid_status?(params[:status]))
+    if (params[:status].present? && !Job.valid_status?(params[:status]))
       render json: { errors: "invalid status given" }, status: 400
       return
+    else
+      status = params[:status].presence
```

Comment:
if you move this up the first if can be `if status && !Job.valid_status?(status)`